### ISSUE TITLE
Trained on GPU, inference on CPU doesn't make sense

### ISSUE THREAD

**Issue by ibeckermayer** [dnouri/nolearn#310]
I trained 3 models using nolearn.lasagne on a GPU (through AWS), but now I am struggling to understand why my model doesn't make any sense now that I'm using it to predict on my CPU.

I was hoping somebody could look at my code and see if there are any obvious errors, before I pay to use a GPU instance again and test if there's a cross compatibility problem.

I train each model, and then save the parameters in a .pkl file using 

```python
model.save_params_to(name)
```

I trained 3 models using AWS, and then transferred the 3 .pkl files with the parameters onto my normal computer. Next, I was hoping to create a dictionary containing all 3 models, but instead of using Conv2DDNNLayer and MaxPool2DDNNLayer, I just use the regular CPU compatible Conv2DLayer and MaxPool2DLayer, and then load_params_from(). My save function, which I ran on my CPU computer, is below: 

```python
def save_specialists():
    # Save a dictionary with all 3 models that can be used to predict
    specialists = OrderedDict()

    model1 = build_model() 
    model2 = build_model() 
    model3 = build_model() 
    model1.output_num_units = 6
    model2.output_num_units = 2
    model3.output_num_units = 4
    model1.load_params_from('1_params.pkl')
    model2.load_params_from('2_params.pkl')
    model3.load_params_from('3_params.pkl')

    specialists['1'] = model1
    specialists['2'] = model2
    specialists['3'] = model3
  
    with open('net-specialists.pickle', 'wb') as f:
        # this time we're persisting a dictionary with all models:
        pickle.dump(specialists, f, -1)
```

Now, I should have a dictionary saved as net-specialists.pickle that contains all 3 of my models, which I can use in my predict function:

```python
def predict(image_name, fname_specialists='net-specialists.pickle'):
    img = cv2.imread(image_name, cv2.IMREAD_GRAYSCALE)
    img = img.reshape(1,1,288,288)
    
    with open(fname_specialists, 'rb') as f:
        specialists = pickle.load(f)

    y_pred = np.empty((1,0))
    for model in specialists.values():
        y_pred1 = model.predict(img)
        y_pred = np.hstack([y_pred, y_pred1])

    y_pred = np.transpose(y_pred)
    print y_pred
    
    plotSample(img,y_pred)
```

However, this does not seem to be working. For example, when I test this on a picture from my training data (which should be overfitted to, at this point), I get the following result:

![example](https://cloud.githubusercontent.com/assets/13578537/20033700/5464d746-a3a7-11e6-84b1-feb98cf62437.png)

As you can see, the points aren't even close (they are supposed to predict key-points in the picture). They are way outside the bounds of my picture. This makes no sense to me, since I've trained a simpler version of this network on my CPU previously (for 10 epochs), using the same functions, and got results that actually made sense. I also don't think I've undertrained the models, since for example my loss for model_1 looks like:

![model1loss](https://cloud.githubusercontent.com/assets/13578537/20033725/eac49852-a3a7-11e6-82c4-17b159c95f20.png)

If somebody sees anything obviously wrong with my methodology, or knows a reason that Conv2DDNNLayer and MaxPool2DDNNLayer cannot be interchanged with Conv2DLayer and MaxPool2DLayer, it would be great to know before I spend more money launching an AWS instance and trying to run the model on a GPU.
